### PR TITLE
use context manager ot open and close nwb files

### PR DESCRIPTION
**Pull request by Saksham20**
currently the nwbextractors keep the nwbfile open.  THis PR uses context manager `NWBHDF5IO` to take care of this

### CODE REVIEW

**Comment by samuelgarcia** [comment]
Hi @Saksham20.
After discussion, @alejoe91 did the very oposite refactoring very recently. See #352

The context manager do not work for extractor because the file need to stay open to retrieve chunks and time vector.
The cost of open/close is high, we cannot afford this at every chunk reading.
So the classic approach of open once and close at the end is more efficient in readers.

**Comment by alejoe91** [comment]
Maybe adding a `__del__` to close the io should do it

**Comment by samuelgarcia** [comment]
I don't think we should play with the `__del__` since it is already implemented at nwb/hdf5 level.
It is beter to leave the low level liobs handle low level close at good timing.

**Comment by alejoe91** [comment]
No, because they makes it fail on windows when we try to delete the file

**Comment by samuelgarcia** [comment]
Lets do a `del  rec` before playing with the `__del__`

**Comment by alejoe91** [comment]
> Lets do a `del rec` before playing with the `__del__`

where? I think that everytime the object is deleted the io should be closed

File: spikeinterface/extractors/nwbextractors.py [hunk 1]
```diff
@@ -265,16 +268,15 @@ def __init__(self, file_path: PathType, electrical_series_name: str = None, samp
         check_nwb_install()
         self._file_path = str(file_path)
         self._electrical_series_name = electrical_series_name
-
-        io = NWBHDF5IO(self._file_path, mode='r', load_namespaces=True)
-        self._nwbfile = io.read()
         
-        timestamps = None
```
Comment (alejoe91) [inline spikeinterface/extractors/nwbextractors.py:272]: @Saksham20 this needs to stay here

File: spikeinterface/extractors/nwbextractors.py [hunk 2]
```diff
@@ -265,16 +268,15 @@ def __init__(self, file_path: PathType, electrical_series_name: str = None, samp
...
+        self.io = NWBHDF5IO(self._file_path, mode='r', load_namespaces=True)
+        self._nwbfile = self.io.read()
         if sampling_frequency is None:
             # defines the electrical series from where the sorting came from
             # important to know the sampling_frequency
             self._es = get_electrical_series(self._nwbfile, self._electrical_series_name)
             # get rate
+            timestamps = None
```
Comment (alejoe91) [inline spikeinterface/extractors/nwbextractors.py:279]: remove

**Review by alejoe91 — CHANGES_REQUESTED** [review]
@Saksham20 same mechanism should be implemented to the SortingExtractor too

**Review by alejoe91 — APPROVED** [review]

**Pull request merged**